primitives/core/src/traits.rs

```diff
@@ -201,6 +202,30 @@ pub trait SpawnNamed: Clone + Send + Sync {
 	///
 	/// The given `name` is used to identify the future in tracing.
 	fn spawn(&self, name: &'static str, future: futures::future::BoxFuture<'static, ()>);
+	/// Spawn the given blocking future.
+	///
+	/// The given `subsystem` and `name` is used to identify the future in tracing.
+	fn spawn_blocking_with_subsystem(
+		&self,
+		name: &'static str,
+		_subsystem: &'static str,
+		future: futures::future::BoxFuture<'static, ()>,
+	) {
+		// Default impl doesn't trace subsystem.
+		self.spawn_blocking(name, future);
+	}
```

Comment:
```suggestion
	);
```

Reply:
I added a default implementation so that I don't break current implementers. Should we force everyone to implement even if they might not have a usescase ?

Reply:
Yeah, I get this, but better to break everybody. This trait is mainly "internal", aka not that many external impls properly.